mne_connectivity/spectral/time.py

```diff
@@ -1,137 +1,275 @@
 
     See Also
     --------
     mne_connectivity.spectral_connectivity_epochs
     mne_connectivity.SpectralConnectivity
-    mne_connectivity.SpectroTemporalConnectivity
+    mne_connectivity.EpochSpectralConnectivity
 
     Notes
     -----
+
+    Please note that the interpretation of the measures in this function
+    depends on the data and underlying assumptions and does not necessarily
+    reflect a causal relationship between brain regions.
+
+    The connectivity measures are computed over time within each epoch and
+    optionally averaged over epochs. High connectivity values indicate that
+    the phase differences between signals stay consistent over time.
+
+    The spectral densities can be estimated using a multitaper method with
+    digital prolate spheroidal sequence (DPSS) windows, or a continuous wavelet
+    transform using Morlet wavelets. The spectral estimation mode is specified
+    using the "mode" parameter.
+
+    By default, the connectivity between all signals is computed (only
+    connections corresponding to the lower-triangular part of the
+    connectivity matrix). If one is only interested in the connectivity
+    between some signals, the "indices" parameter can be used. For example,
+    to compute the connectivity between the signal with index 0 and signals
+    "2, 3, 4" (a total of 3 connections) one can use the following::
+
+        indices = (np.array([0, 0, 0]),    # row indices
+                   np.array([2, 3, 4]))    # col indices
+
+        con = spectral_connectivity_time(data, method='coh',
+                                         indices=indices, ...)
+
+    In this case con.get_data().shape = (3, n_freqs). The connectivity scores
+    are in the same order as defined indices.
+
+    **Supported Connectivity Measures**
+
+    The connectivity method(s) is specified using the "method" parameter. The
+    following methods are supported (note: ``E[]`` denotes average over
+    epochs). Multiple measures can be computed at once by using a list/tuple,
+    e.g., ``['coh', 'pli']`` to compute coherence and PLI.
+
+        'coh' : Coherence given by::
+
+                     | E[Sxy] |
+            C = ---------------------
+                sqrt(E[Sxx] * E[Syy])
+
+        'plv' : Phase-Locking Value (PLV) :footcite:`LachauxEtAl1999` given
+        by::
+
+            PLV = |E[Sxy/|Sxy|]|
+
+        'sxy' : Cross spectrum Sxy
+
+        'pli' : Phase Lag Index (PLI) :footcite:`StamEtAl2007` given by::
+
+            PLI = |E[sign(Im(Sxy))]|
+
+        'wpli' : Weighted Phase Lag Index (WPLI) :footcite:`VinckEtAl2011`
+        given by::
+
+                      |E[Im(Sxy)]|
+            WPLI = ------------------
+                      E[|Im(Sxy)|]
+
+    Parallel computation can be activated by setting the ``n_jobs`` parameter.
+    Under the hood, this utilizes the ``joblib`` library. For effective
+    parallelization, you should activate memory mapping in MNE-Python by
```

Comment:
Perhaps we should add a reasoning to support the "you should activate memory mapping.." statement.

IIUC based on our convo, it is because the total RAM usage shoots up when you consider high-dim variables (i.e. atlas space)? Is that right?

Reply:
The reasoning is not total RAM usage, but rather that when `MNE_MEMMAP_MIN_SIZE=None`, the underlying `joblib` implementation results in pickling and unpickling the whole array each time a pair of indices is accessed.

Reply:
Regarding the size of memory maps, it should be set to something in range 1-100M. Both disabling memory mapping and setting the threshold too high lead to the pickling overhead issue. The experiments were done with example data from MNE, so this might still vary depending on the size of the data. The value 10M seems to work fine for my own analyses of the Cam-CAN data though.